src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -306,25 +308,41 @@ define(function (require, exports, module) {
     function checkAnalyticsDataSend(event, Eventparams, forceSend) {
         var result         = new $.Deferred(),
             isHDTracking   = prefs.get("healthDataTracking"),
-            isEventDataAlreadySent;
+            isEventDataAlreadySent,
+            url = brackets.config.healthDataServerURL,
+            sendFailed = false;
 
         var options = {
             location: {
                 scope: "default"
             }
         };
 
+        var handleFail = function handleFail() {
+            if(sendFailed) {
+                PreferencesManager.setViewState(Eventparams.eventName, 0, options);
+            } else {
+                sendFailed = true;
```

Comment:
What is the use of this sendFailed? Looks like it will always execute else part.